### DISCUSSION TITLE
Since 9.4.0 an extra keybinding needed to overwrite the binding for lazygit

### DISCUSSION THREAD

**Discussion by mhersson** [LazyVim/LazyVim#1641] [Q&A]
Hi.

I use [neogit](https://github.com/NeogitOrg/neogit) instead of lazygit.
Up to version 9.3.1 of LazyVIM I had to only defined the keybinding for neogit like this, and it would "overwrite" lazygit
``` 
  {
    "NeogitOrg/neogit",
    dependencies = {... },
    keys = {
      {
        "<leader>gg",
        "<cmd>Neogit<cr>",
        desc = "Neogit",
      },
    },
.....
    },
  },

```

but from 9.4.0 and on wards I need to also add this keybinding to my keymaps file
```
vim.keymap.set("n", "<leader>gg", require("neogit").open, opts)
```
Only adding it to my keymaps file is enough, but then it shows up without description in which-key.

Is this intended behavior, or is there something I could change so that I can stick to only defining the new keybinding with my neogit config?

Thanks!

**Comment by folke** [comment]
This has been fixed already in lazy.nvim, so please upgrade

**Comment by shelper** [comment]
Hey, @mhersson  do you have issue using Neogit where always a popup window asking for overwriting COMMIT_EDITMSG when committing with noegit?
I reported the issue to neogit at https://github.com/NeogitOrg/neogit/discussions/1473 but seems it is not neogit issue, it is limited to layzvim, as if i put below config in `lua/config/lazy.lua`, no popup, but once i uncomment line `{ "LazyVim/LazyVim", import = "lazyvim.plugins" }`, i got this popup every time

```
require("lazy").setup({
	spec = {
		-- { "LazyVim/LazyVim", import = "lazyvim.plugins" },
	        { "NeogitOrg/neogit", dependencies = { "nvim-lua/plenary.nvim" }, config = true },
	},
...
})
```
> I use [neogit](https://github.com/NeogitOrg/neogit) instead of lazygit. Up to version 9.3.1 of LazyVIM I had to only defined the keybinding for neogit like this, and it would "overwrite" lazygit

**Reply by dpetka2001** [reply]
I just tested out `neogit` and made a commit in my dotfiles and it doesn't show the pop up message. It just commits directly. This looks to be a problem with some "dirty" `.git` on your end.

**Reply by dpetka2001** [reply]
I just created a file `~/.config/nvim/lua/plugins/neogit.lua` with contents
```lua
return {
  {
    "NeogitOrg/neogit",
    cmd = "Neogit",
    opts = {},
  },
}
```
and it worked as expected.

**Reply by shelper** [reply]
it is caused by autocmd i customized.
